Haskell/Haskell.sublime-syntax

```diff
@@ -1,407 +1,1388 @@
-    - match: \b(?:data)\b
-      scope: keyword.declaration.data.haskell
-    - match: \b(?:deriving)\b
-      scope: keyword.declaration.data.haskell
-    - match: \b(?:case|of)\b
+    - match: (?:case|of){{break}}
       scope: keyword.control.conditional.select.haskell  # the construct is commonly called "select"
-    - match: \b(?:let|where)\b
+    - match: if{{break}}
+      scope: keyword.control.conditional.if.haskell
+    - match: then{{break}}
+      scope: keyword.control.conditional.then.haskell
+    - match: else{{break}}
+      scope: keyword.control.conditional.else.haskell
+    - match: let{{break}}
       scope: keyword.declaration.variable.haskell
-    - match: \breturn\b
-      scope: keyword.control.flow.return.haskell
-    - match: \b(?:default|otherwise)\b
-      scope: keyword.other.haskell
-    - match: \b(?:(if)|(then)|(else))\b
-      captures:
-        1: keyword.control.conditional.if.haskell
-        2: keyword.control.conditional.then.haskell
-        3: keyword.control.conditional.else.haskell
-
-  operator:
-    - match: (`)[a-zA-Z_'.\d]+(`)
-      # Haskell allows any ordinary function application (elem 4 [1..10])
-      # to be rewritten as an infix expression (4 `elem` [1..10])."
-      scope: keyword.operator.function.infix.haskell
-      captures:
-        1: punctuation.definition.function.begin.haskell
-        2: punctuation.definition.function.end.haskell
-    - match: '\binfix[lr]?\b'
+    - match: proc{{break}}
+      scope: keyword.declaration.function.haskell
+    # 4.4.2 Fixity Declarations
+    - match: infix[lr]?{{break}}
+      scope: keyword.declaration.fixity.haskell
+
+  operators:
+    - include: sequence-separators
+    - include: big-arrow-operators
+    - include: left-arrow-operators
+    - include: right-arrow-operators
+    - include: prefix-parens-operators
+    - include: infix-quoted-operators
+    # Match all not otherwise matched single quotes as promoition operator
+    # Note: Found in real world code but not in specs so far.
+    - match: \'
       scope: keyword.operator.haskell
-    - match: \b(\d+)(?:(\.)(\d+)([eE][-+]?\d+)?|([eE][-+]?\d+))\b
-      scope: meta.number.float.decimal.haskell
-      captures:
-        1: constant.numeric.value.haskell
-        2: punctuation.separator.decimal.haskell
-        3: constant.numeric.value.haskell
-        4: constant.numeric.value.exponent.haskell
-        5: constant.numeric.value.exponent.haskell
-    - match: '{{operator_infix}}'
+    - match: '{{operator_double_colon}}'
+      scope: keyword.operator.double-colon.haskell
+    - match: '{{operator_symbol}}'
       scope: keyword.operator.haskell
-    - include: operator_paren
 
-  operator_paren:
-    - match: \(({{operator_parens}})\)
-      scope: variable.function.infix.haskell
-      captures:
-        1: keyword.operator.haskell
+  big-arrow-operators:
+    - match: '{{operator_big_arrow}}'
+      scope: keyword.operator.big-arrow.haskell
```

Comment:
While big arrows are (afaik) only used in function signatures, I find it hard to determine the correct scope for them. They are separators, but like the normal arrows, they separate syntactically different segments where it would probably make sense to reflect that in a different scope name that doesn't differ in the 4th subscope.

Reply:
We could use `punctuation.separator`. That's what Erlang does at least.

Reply:
... but all those changes towards `punctuation.separator` cause the syntax to look quite plain in Monokai, which might result in a need to tweak in order to not loose reputation.